bluetooth: fast_pair: add use case layer and Kconfig selection (#16443)
nrfconnect/sdk-nrf: doc/nrf/external_comp/bt_fast_pair.rst

In the official `Fast Pair`_ documentation, the `Fast Pair Device Feature Requirements`_ category defines additional requirements for each supported use case, and specifies a list of mandatory, optional, and unsupported Fast Pair features.
If your product is targeting one of the listed use cases, you must align your accessory firmware to meet these requirements.

If your Google Fast Pair use case is supported by the |NCS|, you can select it using the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE` Kconfig choice option.

> The selected Kconfig option configures the Fast Pair feature and extensions to satisfy the `Fast Pair Device Feature Requirements`_ for your target use case.

Reviewer: Consider explicitly mentioning that some of the requirements need to be implemented at application level.

Reviewer: `features and extensions`

The :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_UNKNOWN` Kconfig option is the

> default choice selection.

Reviewer: One sentence per line

In the official `Fast Pair`_ documentation, the `Fast Pair Device Feature Requirements`_ category defines additional requirements for each supported use case, and specifies a list of mandatory, optional, and unsupported Fast Pair features.
If your product is targeting one of the listed use cases, you must align your accessory firmware to meet these requirements.

> If your Google Fast Pair use case is supported by the |NCS|, you can select it using the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE` Kconfig choice option.

Reviewer 2: Maybe we should mention currently supported use cases?

Author: They are listed in subsequent sections (currently only the "Locator Tag"):

```In this case, you can find more specific description in the following subsections that describe Fast Pair use cases supported by the |NCS|.```

Author: Hmm, I could add a sentence:

```
You can select one of the following use case options:

* :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_UNKNOWN`: the Unknown use case (the default choice) - see the :ref:`ug_bt_fast_pair_use_case_unknown` subsection for more information.
* :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_LOCATOR_TAG`: the Locator Tag use case - see  the :ref:`ug_bt_fast_pair_use_case_locator_tag` subsection for more information.
```
What do you think about it?

Reviewer: Looking at how doc renders, the list of use-cases is generated in side panel located at the right: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/external_comp/bt_fast_pair.html. I doubt if it's necessary to add the list explicitly also here, but I would not block because of it.

Reviewer 2: You have to know that it is there - there is a lot of other items on that list, and also you have to click dropdown button. I wouldn't relay on that.

Author: applied this suggestion

> If your Google Fast Pair use case is supported by the |NCS|, you can select it using the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE` Kconfig choice option.

Reviewer 3: This could say something along the lines: ``Select an option depending on your use case:``
I'd also say agree that the doc structure makes listing these references unnecessary, but at the same time the article as a whole is pretty overwhelming, so it definitely wouldn't hurt to help user navigate

The selected Kconfig option configures the Fast Pair features and extensions to satisfy the `Fast Pair Device Feature Requirements`_ for your target use case.
For certain device types, you may need to implement some of these requirements at the application level.
In this case, you can find more specific description in the following subsections that describe Fast Pair use cases supported by the |NCS|.

The :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_UNKNOWN` Kconfig option is the default selection for the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE` Kconfig choice option.

> This use case configuration is neutral, which means that it does not enable any Fast Pair features and extensions, and it does not impose any restrictions on the Fast Pair Kconfig options.

Reviewer 3: ```suggestion
This use case configuration is neutral, which means it does not enable any Fast Pair features and extensions or impose restrictions on Fast Pair Kconfig options.
```

You can use the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_UNKNOWN` Kconfig option to implement use cases that are not yet supported by the |NCS|.

> In this case, you need to enable the necessary Fast Pair features and extensions explicitly in the application Kconfig configuration.

Author: @MarekPieta, are you ok with this change as it was sort of triggered by the following comment: https://github.com/nrfconnect/sdk-nrf/pull/16443#discussion_r1680469618

> In this case, you need to enable the necessary Fast Pair features and extensions explicitly in the application Kconfig configuration.

Reviewer: `application configuration` sounds ok too. I just wanted to ensure that user is aware that this needs to be applied in configuration and that there is no need for application to implement anything.

Alternatively we can call it `in Fast Pair Kconfig configuration` (following @zycz understanding).

> In this case, you need to enable the necessary Fast Pair features and extensions explicitly in the application Kconfig configuration.

Author: prj.conf is also part of the Kconfig configuration

there is a lot of configuration types: memory configuration, DTS configuration, etc.

> In this case, you need to enable the necessary Fast Pair features and extensions explicitly in the application Kconfig configuration.

Reviewer: Of course, for me we can also leave it as is

> In this case, you need to enable the necessary Fast Pair features and extensions explicitly in the application Kconfig configuration.

Reviewer 3: ```suggestion
In this case, you must manually enable the required Fast Pair features and extensions in the application's Kconfig configuration.
```

In the official `Fast Pair`_ documentation, the `Fast Pair Device Feature Requirements`_ category defines additional requirements for each supported use case, and specifies a list of mandatory, optional, and unsupported Fast Pair features.
If your product is targeting one of the listed use cases, you must align your accessory firmware to meet these requirements.

> You can select an option depending on your use case from the following list that is part of the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE` Kconfig choice:

Reviewer 3: ```suggestion
Depending on your use case, select an option from the following list that is a part of the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE` Kconfig choice:
```

> * :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_UNKNOWN`: the Unknown use case (the default choice) - see the :ref:`ug_bt_fast_pair_use_case_unknown` subsection for more information.

Reviewer 3: ```suggestion
* :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_UNKNOWN` - The unknown use case (the default choice).
  See the :ref:`ug_bt_fast_pair_use_case_unknown` subsection for more information.
```
to limit the amount of different punctuation
Also, not sure if ``Unknown`` should be capitalized (the Kconfig description doesn't stick to uppercase either)

> * :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_LOCATOR_TAG`: the Locator Tag use case - see the :ref:`ug_bt_fast_pair_use_case_locator_tag` subsection for more information.

Reviewer 3: ```suggestion
* :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_LOCATOR_TAG` - The Locator Tag use case.
  See the :ref:`ug_bt_fast_pair_use_case_locator_tag` subsection for more information.
```

Author: I just noticed that the locator tag should not be capitalized either (we do not capitalize it elsewhere).

It is even possible to combine multiple technologies in a single product to improve the user experience.

The `Fast Pair Device Feature Requirements for Locator Tags`_ documentation defines the Fast Pair requirements for the locator tag use case.

> If your product is targeting the locator tag use case, you must manually configure your application according to these requirements in your project configuration file.

Reviewer: I would not inform about possibility of manual configuration. I think we should suggest only to rely on `CONFIG_BT_FAST_PAIR_USE_CASE_LOCATOR_TAG` to simplify doc here

It is even possible to combine multiple technologies in a single product to improve the user experience.

The `Fast Pair Device Feature Requirements for Locator Tags`_ documentation defines the Fast Pair requirements for the locator tag use case.

> If your product is targeting the locator tag use case, you must enable the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_LOCATOR_TAG` Kconfig option that automatically selects the appropriate Fast Pair configuration according to these requirements.

Reviewer: `to use case requirements`?

> This Kconfig option, intended for the locator tag use case, enables the mandatory Fast Pair features and extensions and restricts the usage of unsupported ones.

Reviewer 3: ```suggestion
This Kconfig option activates the necessary Fast Pair features and extensions while restricting the unsupported ones.
```

It is even possible to combine multiple technologies in a single product to improve the user experience.

The `Fast Pair Device Feature Requirements for Locator Tags`_ documentation defines the Fast Pair requirements for the locator tag use case.

> If your product is targeting the locator tag use case, you must enable the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_LOCATOR_TAG` Kconfig option that automatically selects the appropriate Fast Pair configuration according to to the use case requirements.

Reviewer 3: ```suggestion
If your product is targeting the locator tag use case, you must enable the :kconfig:option:`CONFIG_BT_FAST_PAIR_USE_CASE_LOCATOR_TAG` Kconfig option that automatically selects the appropriate Fast Pair configuration according to the use case requirements.
```
missed it in the previous review
